### DISCUSSION TITLE
branch of a branch

### DISCUSSION THREAD

**Discussion by JackCaster** [ropensci/targets#854] [Help]
Hey, I am new to `targets` but I think I got most of my analysis pipeline to work. One thing is missing, however. I need to branch out another branch. For example, with the targets:
```r
list(
  tar_target(x, seq(1, 5), iteration = "list"),
  tar_target(y, seq(1, x), map(x), iteration = "list"), # first branch
  tar_target(z, y * 2, map(y), iteration = "list"), # second branch
  tar_target(p, print(z), iteration = "list")
)
```
I get
```
$z_d0697b30
[1] 2

$z_959da4d3
[1] 2 4

$z_92be5b5f
[1] 2 4 6

$z_f698fbf4
[1] 2 4 6 8

$z_80b956fa
[1]  2  4  6  8 10
```
instead, I would like `z` to create a new branch for each element of the output of `y`. Something like
```
$z_xxxx
[1] 2

$z_xxxx
[1] 2

$z_xxxx
[1] 4

$z_xxxx
[1] 2

$z_xxxx
[1] 4
...
```

Is it possible to achieve something like this? My pipeline is a bit more complex, and the idea is to process different input datasets, and then for each row of each dataset fit a new model.

**Comment by wlandau** [comment]
It is not possible to *directly* dynamically branch over the elements within a dynamic branch, but there are a couple ways you could work around this. One is to combine all the elements of $y$ together first.

```r
# _targets.R
library(targets)
list(
  tar_target(x, seq(1, 5), iteration = "list"),
  tar_target(y, seq(1, x), map(x), iteration = "list"),
  tar_target(y_elements, unlist(y)),
  tar_target(z, y_elements * 2, map(y_elements), iteration = "list")
)
```

Another is to use dynamic branching within static branching (https://books.ropensci.org/targets/static.html#dynamic-within-static-branching). Ultimately, it will probably require rethinking the problem from another angle.
